### ISSUE TITLE
[Question] How can I use AnglELoss, Contrastive Loss and MNRL with no label input  

### ISSUE THREAD

**Issue by dohuyduc2002** [UKPLab/sentence-transformers#3029]
I have a contrastive sentence dataset containing {query:str, positive:str,negative:str} and want to fine-tune a pre-trained embedding model. The authors in the AoE paper have proposed a new approach that enhances the sentence's embedding but only pairs with {query: str, positive: str}. 

I have read the Sbert Losses documentation and discovered that AnglELoss was only suitable for pairwise datasets. I want to experiment with the triplets dataset to test out if the triplets can benefit from multiple losses combination.

**Comment by pesuchin** [comment]
Hi!

It would be a temporary solution, but how about copying AnglELoss to create a custom loss, then modifying that custom loss so it only accepts query and positive?

**Comment by pesuchin** [comment]
Hi, @dohuyduc2002 

The only losses that support triplets are listed in the following URL: MultipleNegativesRankingLoss, CachedMultipleNegativesRankingLoss, TripletLoss, CachedGISTEmbedLoss, and GISTEmbedLoss. 
https://sbert.net/docs/sentence_transformer/loss_overview.html#loss-overview

If you’re set on using AnglELoss, although I can't guarantee it will work effectively, one possible approach is to convert your triplet dataset into pairs. For instance, you could create a dataset that pairs query with positive and query with negative, assigning a similarity score of 1 for the query and positive pairs, and a similarity score of 0 for the query and negative pairs.

To explain more concretely, let’s say you have a triplet dataset as follows:

```
[before]
query, positive, negative
===
query1, positive1, negative1
query2, positive2, negative2
->
[converted]
query, positive, similarity score
===
query1, positive1, 1.0
query1, negative1, 0.0
query2, positive2, 1.0
query2, negative2, 0.0
```

This setup might allow you to use AnglELoss, though the effectiveness of this approach is uncertain.
